src/geotop/matrixview.h

```diff
@@ -0,0 +1,35 @@
+//
+// Created by elisa on 2018/07/27.
+//
+
+#ifndef GEOTOP_MATRIXVIEW_H
+#define GEOTOP_MATRIXVIEW_H
+
+#include "geotop_asserts.h"
+#include "rowview.h"
+
+template <class T> class MatrixView {
+public:
+    T *elem;
+    std::size_t nrh;
+    std::size_t nrl;
+    std::size_t nch;
+    std::size_t ncl;
+
+   RowView<T> operator[](const std::size_t i) noexcept {
+       return RowView<T>{elem, nch, ncl};
+   }
+   const RowView<T> operator[](const std::size_t i) const noexcept {
+       return RowView<T>{elem, nch, ncl};
+   }
+
+    T &operator()(const std::size_t i, const std::size_t j) noexcept {
+        return elem[(i-nrl)*(nch-ncl+1) + (j-ncl)];
```

Comment:
sorry for lot of comments, but you are not doing any range check inside here. Is that correct?

Reply:
This works:
```
 const T &operator()(const std::size_t i, const std::size_t j) const noexcept {
        return (*this)[(i-nrl)*(nch-ncl+1) + (j-ncl)];
```
but this doesn't work:
```
 T &operator()(const std::size_t i, const std::size_t j) noexcept {
    return (*this)[(i-nrl)*(nch-ncl+1) + (j-ncl)];
```
since it says:
```
error: invalid initialization of non-const reference of type ‘double&’ from an rvalue of type ‘RowView<double>’
```

Reply:
anyway I understood why we weren't using ```operator[]```, thank you.

Reply:
I'm not doing any range check so maybe I could add them... tomorrow :)